ballerina/client.bal

```diff
@@ -13,3 +16,4431 @@
+#     
+# 
+# ---
+public isolated client class Client {
+    final http:Client clientEp;
+    final readonly & ApiKeysConfig apiKeyConfig;
+    # Gets invoked to initialize the `connector`.
+    #
+    # + apiKeyConfig - API keys for authorization 
+    # + config - The configurations to be used when initializing the `connector` 
+    # + serviceUrl - URL of the target service 
+    # + return - An error if connector initialization failed 
+    public isolated function init(ApiKeysConfig apiKeyConfig, string serviceUrl, ConnectionConfig config =  {}) returns error? {
+        http:ClientConfiguration httpClientConfig = {httpVersion: config.httpVersion, timeout: config.timeout, forwarded: config.forwarded, poolConfig: config.poolConfig, compression: config.compression, circuitBreaker: config.circuitBreaker, retryConfig: config.retryConfig, validation: config.validation};
+        do {
+            if config.http1Settings is ClientHttp1Settings {
+                ClientHttp1Settings settings = check config.http1Settings.ensureType(ClientHttp1Settings);
+                httpClientConfig.http1Settings = {...settings};
+            }
+            if config.http2Settings is http:ClientHttp2Settings {
+                httpClientConfig.http2Settings = check config.http2Settings.ensureType(http:ClientHttp2Settings);
+            }
+            if config.cache is http:CacheConfig {
+                httpClientConfig.cache = check config.cache.ensureType(http:CacheConfig);
+            }
+            if config.responseLimits is http:ResponseLimitConfigs {
+                httpClientConfig.responseLimits = check config.responseLimits.ensureType(http:ResponseLimitConfigs);
+            }
+            if config.secureSocket is http:ClientSecureSocket {
+                httpClientConfig.secureSocket = check config.secureSocket.ensureType(http:ClientSecureSocket);
+            }
+            if config.proxy is http:ProxyConfig {
+                httpClientConfig.proxy = check config.proxy.ensureType(http:ProxyConfig);
+            }
+        }
+        http:Client httpEp = check new (serviceUrl, httpClientConfig);
+        self.clientEp = httpEp;
+        self.apiKeyConfig = apiKeyConfig.cloneReadOnly();
+        return;
+    }
+
+    # Accepts a cancellation request
+    #
+    # + fulfillmentOrderId - The ID of the fulfillment order
+    # + headers - Headers to be sent with the request 
+    # + return - Accepts a cancellation request sent to a fulfillment service and updates the fulfillment order / Accepts a cancellation request sent to a fulfillment service and updates the fulfillment order 
+    remote isolated function acceptsACancellationRequest(string fulfillmentOrderId, CancellationRequestAcceptJsonBody payload, map<string|string[]> headers = {}) returns CancellationResponse|error {
```

Comment:
What is the reason to use remote methods over resource methods?

Reply:
AFAIR the spec has some complex resource paths like `/admin/api/2025-01/application_charges/{applicationChargeId}.json` which cannot be properly handled by resource functions

Reply:
@SachinAkash01 Can you confirm this?

Reply:
@ThisaruGuruge, As Nuvindu mentioned, this spec includes paths with parameterized path segments(complex paths).
For example:
`'/admin/api/2025-01/storefront_access_tokens/{storefrontAccessTokenId}.json'`

 In such cases, Ballerina syntax does not allow us to map resource functions directly. Therefore, we fall back to generating remote functions instead. 
reference: https://github.com/ballerina-platform/ballerina-spec/issues/1238#issuecomment-1767582117

Reply:
Ack